src/VrRetreat.WebApp/Controllers/AccountController.cs

```diff
@@ -108,4 +108,100 @@ public async Task<IActionResult> Logout()
         await _signInManager.SignOutAsync();
         return RedirectToAction(nameof(HomeController.Index), "Home");
     }
+
+    [HttpGet]
+    public IActionResult Settings()
+    {
+        if (!_signInManager.IsSignedIn(User))
+        {
+            return RedirectToAction("Index", "Home");
+        }
+        
+        return View();
+    }
+
+    [HttpPost]
+    [ValidateAntiForgeryToken]
+    public async Task<IActionResult> ResetPassword(UserSettingsModel settingsModel)
+    {
+        if (!ModelState.IsValid)
+        {
+            return View(nameof(Settings));
+        }
+
+        var currentUser = await _userManager.GetUserAsync(HttpContext.User);
+        var result = await _userManager.ChangePasswordAsync(currentUser, settingsModel.CurrentPassword, settingsModel.Password);
+
+        if (!result.Succeeded)
+        {
+            foreach (var error in result.Errors)
+            {
+                ModelState.TryAddModelError(error.Code, error.Description);
+            }
+            return View(nameof(Settings));
+        }
+
+        await _signInManager.SignOutAsync();
+        return RedirectToAction(nameof(HomeController.Index), "Home");
+    }
+
+    public IActionResult RelinkAccount()
+    {
+        return RedirectToAction(nameof(HomeController.ClaimVrChatName), "Home");
+    }
+
+    public async Task<IActionResult> DownloadAccount()
+    {
+        var currentUser = await _userManager.GetUserAsync(HttpContext.User);
+
+        var dataToDownload = new DownloadData
+        {
+            VrChatId = currentUser.VrChatId,
+            VrChatName = currentUser.VrChatName,
+            VrChatAvatarUrl = currentUser.VrChatAvatarUrl,
+            VrChatLastLogin = currentUser.VrChatLastLogin,
+            UserName = currentUser.UserName,
+        };
+
+        string jsonString = JsonSerializer.Serialize(dataToDownload, new JsonSerializerOptions { WriteIndented = true });
+
+        return File(Encoding.UTF8.GetBytes(jsonString), "application/json;charset=UTF-8");
+    }
+
+    public async Task<IActionResult> UnlinkAccount()
+    {
+        if (!ModelState.IsValid)
+        {
+            return View(nameof(Settings));
+        }
+
+        var currentUser = await _userManager.GetUserAsync(HttpContext.User);
+        currentUser.ClearVrChatLink();
```

Comment:
I don't think this will be enough to update the entity.

I think you also need to inject `IUserRepository` and after you modify the entity, you'd do a:
```cs
userRepository.Update(currentUser);
```